backend/app/tests/api/routes/test_users.py

```diff
@@ -372,42 +372,53 @@ def test_update_user_email_exists(
     assert r.json()["detail"] == "User with this email already exists"
 
 
-def test_delete_user_super_user(
-    client: TestClient, superuser_token_headers: dict[str, str], db: Session
-) -> None:
+def test_delete_user_me(client: TestClient, db: Session) -> None:
     username = random_email()
     password = random_lower_string()
     user_in = UserCreate(email=username, password=password)
-    user = crud.create_user(session=db, user_create=user_in)
-    user_id = user.id
+    _ = crud.create_user(session=db, user_create=user_in)
+
+    login_data = {
+        "username": username,
+        "password": password,
+    }
+    r = client.post(f"{settings.API_V1_STR}/login/access-token", data=login_data)
+    tokens = r.json()
+    a_token = tokens["access_token"]
+    headers = {"Authorization": f"Bearer {a_token}"}
+
     r = client.delete(
-        f"{settings.API_V1_STR}/users/{user_id}",
-        headers=superuser_token_headers,
+        f"{settings.API_V1_STR}/users/me",
+        headers=headers,
     )
     assert r.status_code == 200
     deleted_user = r.json()
     assert deleted_user["message"] == "User deleted successfully"
```

Comment:
```suggestion
    assert deleted_user["message"] == "User deleted successfully"
    result = db.get(User, user_id)
    assert result is None
```

Here we can check that the user by the `user_id` that was created before no longer exists in the DB.

I's useful to test it here because it could happen that we could have some endpoint that was returning the modified data but didn't save the changes in the DB (it happened recently :sweat_smile: ).

This requires importing `User` from `app.models` to use it.

Reply:
I encountered this error running this test "assert <sqlalchemy.engine.result.ScalarResult object at 0x755e21172c50> is None".

I was searching and I found that the result object was not None as expected, but an instance of ScalarResult. This was causing the assertion to fail. In the docs, they mention [one_or_none method](https://docs.sqlalchemy.org/en/20/core/connections.html#sqlalchemy.engine.MappingResult.one_or_none), which fetches a single result from the query if available, or None if no results were found, which aligns with our expectations in this test case, I think :grimacing:

Reply:
I also updated some status codes :sweat_smile:

Reply:
This test was breaking so I had to updated it, also updated some status codes. :sweat_smile: